lib/ansible/executor/task_executor.py

```diff
@@ -372,12 +373,17 @@ def _run_loop(self, items):
                     'msg': 'Failed to template loop_control.label: %s' % to_text(e)
                 })
 
+            # if plugin is loaded, get resolved name, otherwise leave original task connection
+            if self._connection and not isinstance(self._connection, string_types):
+                task_fields['connection'] = getattr(self._connection, 'ansible_name')
```

Comment:
Is it possible we can attempt to ensure we get a FQCN here? Will that already be the case for plugins actually coming from collections?

Reply:
that is what ansible_name is supposed to reflect